ner/src/main/java/edu/illinois/cs/cogcomp/ner/ExpressiveFeatures/GazetteersFactory.java

```diff
@@ -37,29 +43,35 @@
      */
     static public void init(int maxPhraseLength, String path, boolean flatgazetteers)
             throws IOException {
-        synchronized (GAZ_INIT_LOCK) {
-            if (gazetteers == null) {
+
+        if(gazetteers_map.containsKey(path)) {
+            gazetteers = gazetteers_map.get(path);
+        } else {
+            synchronized (GAZ_INIT_LOCK) {
                 if (flatgazetteers) {
                     gazetteers = new FlatGazetteers(path);
                 } else {
                     gazetteers = new TreeGazetteers(maxPhraseLength, path);
                 }
-            } else {
```

Comment:
Because there might be language specific gazetteers, and sometimes different languages share the same set of gazetteers, we don't want to load them again.